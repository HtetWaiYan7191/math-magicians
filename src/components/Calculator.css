* {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
}
.output-result {
    grid-column: 1/5;
    text-align: end;
}

.container {
    margin: 0 auto;
    width: 50%;
    display: grid;
    grid-template-columns: repeat(4,1fr);
    background-color: yellow;
    text-align: center;
}

.container:last-child {
    grid-column: 2/5;
}