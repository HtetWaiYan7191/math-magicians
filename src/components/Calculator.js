import React from 'react';
import './Calculator.css';
import Buttons from './Buttons';

const numbers = ['AC', '+/-', '%', '\u00F7', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '='];
const Calculator = () => (
  <div className="container">
    <div className="output-result">0</div>
    {numbers.map((number) => <Buttons number={number} key={number} />)}
  </div>
);

export default Calculator;
