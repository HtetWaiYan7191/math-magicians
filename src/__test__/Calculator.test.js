import React from "react";
import renderer from 'react-test-renderer'
import Calculator from '../components/Calculator';

describe('Calculator', () => {
    test('should render the component', () => {
        const component = renderer.create(<Calculator/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})