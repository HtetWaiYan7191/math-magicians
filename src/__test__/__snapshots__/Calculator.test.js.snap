// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render the component 1`] = `
<div
  className="container"
>
  <div
    className="output-result"
  >
    0
  </div>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="input-button orange"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="input-button orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="input-button orange"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="input-button orange"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="input-button zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="input-button"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="input-button orange"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
